app/main.py

```diff
@@ -2,0 +1,64 @@
+class SoftwareEngineer:
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class CreatorWebPageMixin(SoftwareEngineer):
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class CreatorApiMixin(SoftwareEngineer):
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class FrontendDeveloper(CreatorWebPageMixin, SoftwareEngineer):
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+        self.create_awesome_web_page()
+
+
+class BackendDeveloper(CreatorApiMixin, SoftwareEngineer):
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
+        self.create_powerful_api()
```

Comment:
Calling `create_powerful_api()` in the constructor of `BackendDeveloper` may not be necessary and could lead to side effects during object initialization. Consider removing this call from the constructor.